### PR TITLE
Data catalog done

### PR DESCRIPTION
**Pull request by sfilipi**
Towards #1209 

Adding a sample for TrainTestSplit. 
Adding doc strings for the CreateFromeEnumerable
Minor fixes to xml formatting.

### CODE REVIEW

File: src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs [hunk 1]
```diff
@@ -82,6 +82,25 @@ public IDataView LoadFromEnumerable<TRow>(IEnumerable<TRow> data, SchemaDefiniti
...
+        /// One typical usage for streaming data view could be: create the data view that lazily loads data
+        /// as needed, then apply pre-trained transformations to it and cursor through it for transformation
+        /// results.
+        /// One practical usage of this would be to supply the feature column names through the <see cref="DataViewSchema.Annotations"/>.
+        /// </remarks>
+        /// <typeparam name="TRow">The <typeparamref name="TRow"/> to convert to an <see cref="IDataView"/>.</typeparam>
+        /// <param name="data">The data with <typeparamref name="TRow"/> to convert to an <see cref="IDataView"/>.</param>
+        /// <param name="schema">The schema of the returned <see cref="IDataView"/>.</param>
+        /// <returns>An <see cref="IDataView"/> with the given <paramref name="schema"/>.</returns>
         public IDataView LoadFromEnumerable<TRow>(IEnumerable<TRow> data, DataViewSchema schema)
```
Comment (rogancarr) [inline src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs]: >LoadFromEnumerable [](start = 25, length = 18)

nit: Whoever add these didn't do so in alphabetical order.

Comment (sfilipi) [reply src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs]: they are added automatically by VS. I think the go in the same order which the arguments appear.

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/TrainTestSplit.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.ML.Data;
+using static Microsoft.ML.DataOperationsCatalog;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    /// <summary>
+    /// Sample class showing how to use ShuffleRows.
```
Comment (zeahmed) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/TrainTestSplit.cs]: >ShuffleRows [](start = 40, length = 11)

ShuffleRows???

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/TrainTestSplit.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    /// </summary>
+    public static class TrainTestSplit
+    {
+        public static void Example()
+        {
+            // Creating the ML.Net IHostEnvironment object, needed for the pipeline.
+            var mlContext = new MLContext();
+
+            // Download and featurize the dataset.
```
Comment (zeahmed) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/TrainTestSplit.cs]: >// Download and featurize the dataset [](start = 12, length = 37)

No download and featurize...:)
Generate random data.

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/TrainTestSplit.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                {
+                    Group = i % 2,
+
+                    // Create random features that are correlated with label.
+                    Features = randomFloat()
+                };
+            }
+        }
+
+        // Example with label and 50 feature values. A data set is a collection of such examples.
```
Comment (zeahmed) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/TrainTestSplit.cs]: >// Example with label and 50 feature values. A data set is a collection of such examples. [](start = 8, length = 89)

copy-paste error???

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/TrainTestSplit.cs [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            var dataview = mlContext.Data.LoadFromEnumerable(examples);
+
+            // Leave out 10% of the dataset for testing.For some types of problems, for example for ranking or anomaly detection,
+            // we must ensure that the split leaves the rows with the same value in a particular column, in one of the splits. 
+            // So below, we specify Group column as the column containing the sampling keys.
+            // Notice how keeping the rows with the same value in the Group column overrides the testFraction definition. 
+            TrainTestData split = mlContext.Data.TrainTestSplit(dataview, testFraction: 0.1, samplingKeyColumnName: "Group");
+
+            var trainDataPreview = split.TrainSet.Preview();
+            var testDataPreview = split.TestSet.Preview();
```
Comment (zeahmed) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/TrainTestSplit.cs]: Can these lines be removed?

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/TrainTestSplit.cs [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+            // [Group, 0], [Features, 0.2060332]
+            // [Group, 1], [Features, 0.4421779]
+            // [Group, 0], [Features, 0.9775497]
+            // [Group, 1], [Features, 0.2737045]
+
+            // The data in the Test split.
+            // [Group, 1], [Features, 0.5588848]
+            // [Group, 0], [Features, 0.9060271]
+
+    }
```
Comment (singlis) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/TrainTestSplit.cs]: tabbing looks off.

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/TrainTestSplit.cs [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+            var random = new Random(seed);
+            float randomFloat() => (float)random.NextDouble();
+            for (int i = 0; i < count; i++)
+            {
+                yield return new DataPoint
+                {
+                    Group = i % 2,
+
+                    // Create random features that are correlated with label.
+                    Features = randomFloat()
```
Comment (singlis) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/TrainTestSplit.cs]: >randomFloat [](start = 31, length = 11)

suggestion: why not replace this with (float)random.NextDouble()?

File: src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs (hunk 1 above)
Comment (sfilipi) [reply src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs]: they are added automatically by VS. I think the go in the same order which the arguments appear. 

---
In reply to: [267126720](https://github.com/dotnet/machinelearning/pull/3021#discussion_r267126720) [](ancestors = 267126720)

**Pull request merged**